src/platform/Linux/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -99,41 +160,37 @@ CHIP_ERROR LinuxWiFiDriver::RevertConfiguration()
     return CHIP_NO_ERROR;
 }
 
-bool LinuxWiFiDriver::NetworkMatch(const WiFiNetwork & network, ByteSpan networkId)
-{
-    return networkId.size() == network.ssidLen && memcmp(networkId.data(), network.ssid, network.ssidLen) == 0;
-}
-
 Status LinuxWiFiDriver::AddOrUpdateNetwork(ByteSpan ssid, ByteSpan credentials, MutableCharSpan & outDebugText,
                                            uint8_t & outNetworkIndex)
 {
     outDebugText.reduce_size(0);
     outNetworkIndex = 0;
-    VerifyOrReturnError(mStagingNetwork.ssidLen == 0 || NetworkMatch(mStagingNetwork, ssid), Status::kBoundsExceeded);
-
-    static_assert(sizeof(WiFiNetwork::ssid) <= std::numeric_limits<decltype(WiFiNetwork::ssidLen)>::max(),
-                  "Max length of WiFi ssid exceeds the limit of ssidLen field");
-    static_assert(sizeof(WiFiNetwork::credentials) <= std::numeric_limits<decltype(WiFiNetwork::credentialsLen)>::max(),
-                  "Max length of WiFi credentials exceeds the limit of credentialsLen field");
+    VerifyOrReturnError(mStagingNetwork.Empty() || mStagingNetwork.Match(ssid), Status::kBoundsExceeded);
 
     // Do the check before setting the values, so the data is not updated on error.
     VerifyOrReturnError(credentials.size() <= sizeof(mStagingNetwork.credentials), Status::kOutOfRange);
-    VerifyOrReturnError(ssid.size() <= sizeof(mStagingNetwork.ssid), Status::kOutOfRange);
+    VerifyOrReturnError(!ssid.empty() && ssid.size() <= sizeof(mStagingNetwork.ssid), Status::kOutOfRange);
 
     memcpy(mStagingNetwork.credentials, credentials.data(), credentials.size());
     mStagingNetwork.credentialsLen = static_cast<decltype(mStagingNetwork.credentialsLen)>(credentials.size());
 
     memcpy(mStagingNetwork.ssid, ssid.data(), ssid.size());
     mStagingNetwork.ssidLen = static_cast<decltype(mStagingNetwork.ssidLen)>(ssid.size());
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+    mStagingNetwork.networkIdentityLen = 0;
+    mStagingNetwork.clientIdentityLen  = 0;
+    mStagingNetwork.clientIdentityKeypair.reset();
+#endif // CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+
     return Status::kSuccess;
 }
 
 Status LinuxWiFiDriver::RemoveNetwork(ByteSpan networkId, MutableCharSpan & outDebugText, uint8_t & outNetworkIndex)
 {
     outDebugText.reduce_size(0);
     outNetworkIndex = 0;
-    VerifyOrReturnError(NetworkMatch(mStagingNetwork, networkId), Status::kNetworkIDNotFound);
+    VerifyOrReturnError(mStagingNetwork.Match(networkId), Status::kNetworkIDNotFound);
```

Comment:
Should maybe be:
```suggestion
    VerifyOrReturnError(mStagingNetwork.Matches(networkId), Status::kNetworkIDNotFound);
```

Reply:
I have no strong opinion on this... I've been trying to discern what the most followed convention in the code base is, since we have e.g. `empty` (rather than `IsEmpty`) and `data_equal`. I personally tend to prefer "Matches" / "Is..."